src/DataLayouts/DataLayouts.jl

```diff
@@ -302,12 +302,20 @@ function replace_basetype(data::AbstractData{S}, ::Type{T}) where {S, T}
     )
 end
 
+maybe_populate!(array, ::typeof(similar)) = nothing
+maybe_populate!(array, ::typeof(ones)) = fill!(array, 1)
+maybe_populate!(array, ::typeof(zeros)) = fill!(array, 0)
+function maybe_populate!(array, ::typeof(rand))
+    parent(array) .= typeof(array)(rand(eltype(array), size(array)))
+end
+
 # ==================
 # Data3D DataLayout
 # ==================
 
 """
     IJKFVH{S, Nij, Nk}(array::AbstractArray{T, 6}) <: Data3D{S, Nij, Nk}
+    IJKFVH{S}(ArrayType[, ones | zeros | rand]; Nij, Nk, Nv)
 
 A 3D DataLayout. TODO: Add more docs
```

Comment:
I think these all look good, but can you please add more documentation? 

It took me a couple of minutes to wrap my head around what they were doing and I had to read the code to understand it.

I think the docstrings should explain what these constructors do and highlight any possible pitfall with using them.

Reply:
Yeah, good point. I added some docs, can you take a second look?